app/src/main/java/org/thoughtcrime/securesms/home/HomeViewModel.kt

```diff
@@ -46,8 +55,18 @@ class HomeViewModel @Inject constructor(
      * This flow will emit whenever the user asks us to reload the conversation list or
      * whenever the conversation list changes.
      */
-    val threads: StateFlow<Data?> = combine(observeConversationList(), observeTypingStatus(), ::Data)
-            .stateIn(viewModelScope, SharingStarted.Eagerly, null)
+    val data: StateFlow<Data?> = combine(
+        observeConversationList(),
+        observeTypingStatus(),
+        messageRequests(),
+        ::Data
+    )
+        .stateIn(viewModelScope, SharingStarted.Eagerly, null)
+
+    private fun hasHiddenMessageRequests() = TextSecurePreferences.events
+        .filter { it == TextSecurePreferences.HAS_HIDDEN_MESSAGE_REQUESTS }
+        .map { prefs.hasHiddenMessageRequests() }
+        .onStart { emit(prefs.hasHiddenMessageRequests()) }
```

Comment:
I think we can run these on IO as well. `SharedPreferences` - they are actually reading from disk? Albeit small and fast it's also very easy to defer it here

Reply:
possibly overkill, as it should generally be loaded into memory, but this one is super easy to do, so... done!